### PR TITLE
iox-#1196 fix warnings in posix access rights

### PR DESCRIPTION
**Pull request by elfenpiff**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Part of #1196

### CODE REVIEW

**Review by MatthiasKillat — DISMISSED** [review]

File: iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/posix_access_rights.hpp [hunk 1]
```diff
@@ -41,21 +41,22 @@ struct PosixRights
 class PosixGroup
 {
   public:
-    using string_t = cxx::string<100>;
+    static constexpr uint64_t MAX_GROUP_NAME_LENGTH = 128;
+    using groupName_t = cxx::string<MAX_GROUP_NAME_LENGTH>;
     explicit PosixGroup(const gid_t f_id) noexcept;
-    explicit PosixGroup(const string_t& f_name) noexcept;
+    explicit PosixGroup(const groupName_t& f_name) noexcept;
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/posix_access_rights.hpp:47]: Not necessary but you could remove the `f_` suffix from all the parameters.

File: iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/posix_access_rights.hpp [hunk 2]
```diff
@@ -41,21 +41,22 @@ struct PosixRights
 class PosixGroup
 {
   public:
-    using string_t = cxx::string<100>;
+    static constexpr uint64_t MAX_GROUP_NAME_LENGTH = 128;
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/posix_access_rights.hpp:44]: Out of curiosity: Why did you choose 128 for the max group name length and 32 for the max user name length?

Comment (elfenpiff) [reply iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/posix_access_rights.hpp:44]: See `man useradd` there it is stated that the max username length is 32.
The maximum length of a group is 128.

Comment (FerdinandSpitzschnueffler) [reply iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/posix_access_rights.hpp:44]: Got it for the user name but can you provide me a link for the max group name length? Maybe I mix something up here but when I look for instance at `groupadd` documentation, it states that group names can only be up to 16 characters long.

File: iceoryx_hoofs/source/posix_wrapper/access_control.cpp [hunk 3]
```diff
@@ -104,50 +103,38 @@ AccessController::createACL(const int32_t numEntries) noexcept
...
-        }
-
-        auto id = posix::PosixUser::getUserID(name);
-        if (!id.has_value())
-        {
-            return false;
-        }
-
-        return addPermissionEntry(category, permission, id.value());
+        std::cerr << "Error: specific users must have an explicit name." << std::endl;
```
Comment (MatthiasKillat) [inline iceoryx_hoofs/source/posix_wrapper/access_control.cpp:110]: Not sure how we want to deal with these kind of errors. Later on we can have something like a "NonEmptyString" and the interface itself will not allow this kind of error.

This is for future designs though, basically have non-null or non-empty type variations of some of the types. I think this one here can even be templated like `NonEmpty\<string\>`. The crucial question here is always how to keep it expressive AND keep checking overhead low.

Nothing to be changed in this PR.

**Review by MatthiasKillat — APPROVED** [review]

**Pull request merged**